dvc/scm/git/tree.py

```diff
@@ -76,10 +76,31 @@ def _is_tree_and_contains(obj, path):
                 return True
         return False
 
+    def _try_fetch_from_remote(self):
+        import git
+
+        try:
+            # checking if tag/branch exists locally
+            self.git.git.show_ref(self.rev, verify=True)
+            return
+        except git.exc.GitCommandError:
+            pass
+
+        try:
+            # checking if it exists on the remote
+            self.git.git.ls_remote("origin", self.rev, exit_code=True)
+            # fetching remote tag/branch so we can reference it locally
+            self.git.git.fetch("origin", "{rev}:{rev}".format(rev=self.rev))
+        except git.exc.GitCommandError:
```

Comment:
should we fail if does not exist?

Reply:
We will fail later down the line, where we will try to use this. I'll take a second look to see if failing here would provide better UX. Thanks for the heads up!

Reply:
Adjusted the error handling down the line to handle it more gracefully.